card_records_spec.rb

```diff
@@ -0,0 +1,294 @@
+        permit_status = subject.send(:browser).select(name: "ctl00$ctl00$_main$main$ddlPermitStatus").value
+        status_year = subject.send(:browser).select(name: "ctl00$ctl00$_main$main$ddlStatusYear").value
+
+        expect(permit_type).to eq "All"
+        expect(permit_status).to eq "All"
+        expect(status_year).to eq current_year
+      end
+    end
+  end
+
+  context "Stage 2" do
+    before(:each) do
+      uri = "#{file_prefix}#{file_fixture("Stage_2.htm")}"
+
+      allow(subject).to receive(:uri).and_return(uri)
+
+      subject.goto(uri)
+    end
+
+    describe "#run" do
+      let(:stage_four_file) { "#{file_prefix}#{file_fixture("Stage_4.htm")}" }
+      let(:count_of_pages) { 50 }
+
+      before(:each) do
+        allow(subject).to receive(:download_pdf)
+        allow(subject).to receive(:upload_pdf)
+        allow(subject).to receive(:goto)
+        allow(subject).to receive(:fill_form)
+        allow(subject).to receive(:submit_form)
+        allow_any_instance_of(Watir::Row).to receive(:link).and_return(double(href: stage_four_file))
+      end
+
+      it "saves all rows and info tables into db" do
+        subject.run
+
+        expect(NewMexicoQueryRecord.count).to eq count_of_pages
+        expect(NewMexicoCardRecord.count).to eq count_of_pages
+      end
+
+      it "saves rows and tables with correct parameters" do
+        subject.run
+
+        nm_query_record = NewMexicoQueryRecord.first
+        nm_query_record_expected_params =
+          {
+            "filing_id" => "269757",
+            "document_type" => "Tubing",
+            "document_comment" => "FASKEN OIL & RANCH LTD[151416]\nDENTON SWD #003\n30-025-05306",
+            "document_status" => "APPROVED",
+            "scan_date" => Date.parse("2019-07-16")
+          }
+
+        nm_card_record = NewMexicoCardRecord.take
+        nm_card_record_expected_params =
+          {
+            "permit_id"=>"269757",
+            "operator_name"=>"FASKEN OIL & RANCH LTD [151416]",
+            "well_name_full"=>"DENTON SWD #003",
+            "location"=>"M-12-15S-37E",
+            "previous_name"=>"",
+            "new_name"=>"",
+          }
+
+        expect(nm_query_record.attributes).to include nm_query_record_expected_params
+        expect(nm_card_record.attributes).to include nm_card_record_expected_params
+      end
+
+      it "uploads all of pdf to S3" do
+        expect(subject).to receive(:upload_pdf).exactly(count_of_pages).times
+
+        subject.run
+      end
+    end
+
+    describe "#count_of_pages" do
+      it "returns count of table pages" do
+        expect(subject.count_of_pages).to eq 2
+      end
+    end
+
+    describe "#rows" do
+      it "returns rows without headers" do
+        headers = ["Id", "Type", "Description", "Status", "Status Date"]
+
+        first_row_text = subject.rows.first.cells.map(&:text)
+
+        expect(first_row_text).to_not eq headers
```

Comment:
Can you change this to a more specific test for the actual values that get returned? Or is this test pretty similar to `cells_text_of_row`?

Reply:
I do not fully understand what you mean.

We just need to ensure that the row with headers does not fall into the rows of the table we need.